lightning/src/ln/channelmanager.rs

```diff
@@ -1383,6 +1414,24 @@ macro_rules! handle_error {
 	}
 }
 
+macro_rules! update_maps_on_chan_removal {
+	($self: expr, $short_to_id: expr, $channel: expr) => {
+		if let Some(short_id) = $channel.get_short_channel_id() {
+			$short_to_id.remove(&short_id);
+		} else {
+			// If the channel was never confirmed on-chain prior to its closure, remove the
+			// outbound SCID alias we used for it from the collision-prevention set. While we
+			// generally want to avoid ever re-using an outbound SCID alias across all channels, we
+			// also don't want a counterparty to be able to trivially cause a memory leak by simply
+			// opening a million channels with us which are closed before we ever reach the funding
+			// stage.
+			let alias_removed = $self.outbound_scid_aliases.lock().unwrap().remove(&$channel.outbound_scid_alias());
+			debug_assert!(alias_removed);
```

Comment:
Not sure what you mean here? We're only removing if `get_short_channel_id` is `None`.

Reply:
I think we should be removing even if it *isn't* `None`, at least pre-0conf?

Reply:
I don't think so? See the comment two lines up - the only reason for this set is to try to ensure we don't re-use SCID aliases. Just because a channel was closed doesn't mean we want to re-use them later, at least until *much* later.